### PR TITLE
Revamp dactyl_manuform directory

### PR DESCRIPTION
**Pull request by dunk2k**
Restructure `dactyl_manuform` directory in line with #21492 

## Description
- moved *_5 variants to `dactyl_manuform_mini` directory
- moved 3 thumb key cluster variants to `dactyl_manuform_tight` directory
- added aliases for variants that have been moved/renamed
- added `readme.md` per variant
- layout aliases added, per variant, for prior layout macro name
- where none existed prior:
    - added `via` keymap per variant
    - added license headers to .c and .h files 
- removed `config.h` files with `MASTER LEFT` defined only 
- removed rgblight definitions as dactyl manuform does not have a standard/convention for them
    - keymaps that used prior rgblight definitions have them added to their respective directory 
 
cc: @l4u - regarding `dactyl_manuform_mini/` directory

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/4x5/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                   _______, _______,                                      _______, _______,
+                                                     _______, _______,  _______, _______,
+                                                     _______, _______,  _______, _______
+    )
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/4x5/keymaps/via/keymap.c:72]: I suspect that this is just being moved (and git sometimes sucks about this), but this function is deprecated and should be removed.  It's also... not being used here.
```suggestion
```

File: keyboards/handwired/dactyl_manuform/4x5_eic/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    _______,      _______, KC_6,    KC_7,    KC_8,    KC_9,    KC_0,
+        KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,                        KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN,
+                 KC_F11,  KC_F12,                                                    _______, _______,
+                                            _______, _______,      _______, _______,
+                                            _______, _______,      _______, _______,
+                                            _______, _______,      _______, _______
+    )
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/4x5_eic/keymaps/default/keymap.c:71]: same

**Review by drashna — COMMENTED** [review]
Mostly because I don't want to tag every keymap, but the `persistent_default_layer_set` function should be removed from all of the default/via keymaps.

File: keyboards/handwired/dactyl_manuform/4x5/keymaps/via/keymap.c (hunk 1 above)
Comment (dunk2k) [reply keyboards/handwired/dactyl_manuform/4x5/keymaps/via/keymap.c]: Amended as per [2374fbe](https://github.com/qmk/qmk_firmware/pull/22261/commits/2374fbe326037b8c2c9a1b704621208c12d8f4bc), thanks @drashna 🙂

**Review by drashna — APPROVED** [review]

**Comment by dunk2k** [comment]
please label as awaiting review

**Comment by hughrawlinson** [comment]
Booo! @dunk2k has been working consistently on getting this historically significant keyboard into a good state. This PR needs to be reviewed, and not left to languish! If it's because the PR is too big, that needs to be stated so that the PR can be broken into smaller pieces.

**Comment by dunk2k** [comment]
@hughrawlinson Thank you for the kind words. I'm willing to make changes to this PR based on collaborator feedback.

**Comment by infinityis** [comment]
I'm not in the QMK collaborator group, but I would recommend picking one "fix" at a time across this set of keyboards and target that with a dedicated PR. This monolithic PR has too much content to review and approve within a single PR, and I'd hate to see all the effort invested fall by the wayside on account of that. But if (for example) you submit a single PR that only "added license headers to .c and .h files" for this same same set of keyboards, then it would likely sail through the PR process pretty quickly. Or a single PR that changes the layer `#defines` to enums...there isn't much to object to and it becomes very easy to review and approve.

Changes that require more discussion and questions (like modifying the wiring) might warrant more scrutiny and take longer to get through, but by isolating such changes in a more focused PR, it becomes much easier/quicker/cleaner to review and approve because it is a more limited, consistent change being applied across a group of keyboards.

**Comment by hughrawlinson** [comment]
Split up into individual PRs like this you mean?

image

**Pull request closed without merging**